src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpUseImplicitTypeDiagnosticAnalyzer.cs

```diff
@@ -151,6 +152,57 @@ protected override bool TryAnalyzeVariableDeclaration(TypeSyntax typeName, Seman
             return false;
         }
 
+        private bool TryAnalyzeDeclarationExpression(
+            DeclarationExpressionSyntax declarationExpression,
+            SemanticModel semanticModel,
+            OptionSet optionSet,
+            CancellationToken cancellationToken)
+        {
+            // It's not always safe to convert a decl expression like "Method(out int i)" to 
+            // "Method(out var i)".  Changing to 'var' may cause overload resolution errors.  
+            // Have to see if using 'var' means not resolving to the same type as before.  
+            // Note: this is fairly expensive, so we try to avoid this if we can by seeing if
+            // there are multiple candidates with the original call.  If not, then we don't
+            // have to do anything.
+            if (declarationExpression.Parent is ArgumentSyntax argument &&
+                argument.Parent is ArgumentListSyntax argumentList &&
+                argumentList.Parent is InvocationExpressionSyntax invocationExpression)
```

Comment:
note: tehre are other circumstances you might see a declaration expression (like in a constructor call).  i.e. "new Foo(a, b, c, out var d)".  However, these should be extremely rare compared to the normal method-call case.  So i'm only bothering putting the optimization in for that case.